src/config.rs

```diff
@@ -0,0 +1,399 @@
+//! OCI Image Configuration
+//!
+//! Definition following <https://github.com/opencontainers/image-spec/blob/v1.0/config.md>
+
+use std::collections::HashMap;
+
+use chrono::{DateTime, NaiveDateTime, SecondsFormat, Utc};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+/// Helper to deserialize a time
+fn primitive_date_time_from_str<'de, D: Deserializer<'de>>(
+    d: D,
+) -> Result<Option<DateTime<Utc>>, D::Error> {
+    let s = <Option<&str>>::deserialize(d)?
+        .ok_or_else(|| serde::de::Error::invalid_length(0, &"<TIME>"))?;
+
+    let ndt = NaiveDateTime::parse_from_str(s, "%+")
+        .map_err(|err| serde::de::Error::custom::<String>(err.to_string()))?;
+
+    Ok(Some(DateTime::<Utc>::from_utc(ndt, Utc)))
+}
+
+/// Helper to serialize a time
+fn serialize_time<S>(time: &Option<DateTime<Utc>>, s: S) -> Result<S::Ok, S::Error>
```

Comment:
`chrono` has a `serde` feature that we should probably use rather than our own serializer/deserializer: https://docs.rs/chrono/latest/chrono/serde/index.html

Reply:
Yes, thanks. Fixed it